internal/users/db/usergroups.go

```diff
@@ -0,0 +1,152 @@
+package db
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/ubuntu/authd/log"
+)
+
+// UserGroups returns all groups for a given user or an error if the database is corrupted or no entry was found.
+func (c *Database) UserGroups(uid uint32) ([]GroupDB, error) {
+	return userGroups(c.db, uid)
+}
+
+func userGroups(db queryable, uid uint32) ([]GroupDB, error) {
+	query := `
+		SELECT g.name, g.gid, g.ugid 
+		FROM users_to_groups ug
+		JOIN groups g ON ug.gid = g.gid
+		WHERE ug.uid = ?
+	`
+	rows, err := db.Query(query, uid)
+	if err != nil {
+		return nil, fmt.Errorf("query error: %w", err)
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			log.Warningf(context.Background(), "failed to close rows: %v", err)
+		}
+	}()
```

Comment:
We've enough of these calls that maybe they can be in a separated function to be just called with `defer closeRows(rows)`?